cmd/reference/internal/deploy.go

```diff
@@ -421,20 +422,30 @@ func (c *Reconciler) toRevision(deployName string, cm *corev1.ConfigMap) (
 		boxcutter.WithPreviousOwners(previous),
 		boxcutter.WithProbe(
 			boxcutter.ProgressProbeType,
-			boxcutter.ProbeFunc(func(obj client.Object) (success bool, messages []string) {
+			boxcutter.ProbeFunc(func(obj client.Object) probing.Result {
 				u, ok := obj.(*unstructured.Unstructured)
 				if obj.GetObjectKind().GroupVersionKind().Kind != "ConfigMap" || !ok {
-					return true, nil
+					return probing.Result{
```

Comment:
would it be worth having `probing.TrueResult` and `probing.FalseResult(msg ...string)` helpers?